---
title: "Primer 37"
author: "Markus Min"
date: "11/1/2018"
output: html_document
---
## Primer 37
##### Forward: CAAGCTCATTCAAGTGCT
##### Reverse: GACCACTTACACAATTCT
##### Forward primer Tm: 53.8 C
##### Reverse primer Tm: 49.9 C
##### Forward GC Content: 44% (8/18 bases)
##### Reverse GC Content: 39% (7/18 bases)
##### Amplicon length: 154 BP

Notes: This primer was created using ecoPrimers, a program used to design primers that amplify a certain group of target taxa and avoid a group of untarget taxa. The input sequences that were used in ecoPrimers were generated by amplifying vertebrate sequences using a primer from a previous study that was used to amplify the Sebastes cytB gene. The target group was the genus Sebastes, and the untarget group was the order Perciformes. 

### Importing Data 
```{r}
library(ROBIUtils)
library(ROBITaxonomy)
library(ROBIBarcodes)
library(ROBITools)

setwd("//Users/Markus/Documents/UCLA/Barber Lab/Honor's Thesis/ecoPCR_results")
fish = read.ecopcr.result("primer37_10.31.18.ecoPCR")
setwd("//Users/Markus/Documents/UCLA/Barber Lab/Honor's Thesis/R_Files/ncbidb")
taxo = read.taxonomy("ncbidatabase")
teleo.taxid = ecofind(taxo, '^Teleostei$')
is_a_fish=is.subcladeof(taxo,fish$taxid,teleo.taxid)
sebastes.taxid = ecofind(taxo, '^Sebastes$')
is_a_rockfish=is.subcladeof(taxo,fish$taxid,sebastes.taxid)
table(is_a_fish)
# 659 fish sequences amplified
```
Notes: The object "fish" is created by reading the file "primer37_10.31.18.ecoPCR," which contains the amplified sequences from using Primer 37 in an in silico PCR through the program "ecoPCR."

### Testing the Conservation of the Priming Sites
```{r}


Sebastes.forward = ecopcr.forward.shanon(ecopcr = fish,
                                     group = is_a_rockfish)
Sebastes.reverse = ecopcr.reverse.shanon(ecopcr = fish,
                                     group = is_a_rockfish)

Fish.forward = ecopcr.forward.shanon(ecopcr = fish,
                                     group = is_a_fish)
Fish.reverse = ecopcr.reverse.shanon(ecopcr = fish,
                                     group = is_a_fish)

par(mfrow=c(2,1))
dnalogoplot(Sebastes.forward$'TRUE',
            primer = "CAAGCTCATTCAAGTGCT",
            main='Forward Sebastes')
dnalogoplot(Sebastes.reverse$'TRUE',
            primer = "GACCACTTACACAATTCT",
            main='Reverse Sebastes')
dnalogoplot(Fish.forward$'TRUE',
            primer = "CAAGCTCATTCAAGTGCT",
            main='Forward Fish')
dnalogoplot(Fish.reverse$'TRUE',
            primer = "GACCACTTACACAATTCT",
            main='Reverse Fish')

```

These plots show the match between the primer and the target sequence for Sebastes and all fish (Teleosts). This illustrates that the primer conservation is much better for Sebastes than for all Teleosts, showing that Sebastes sequences will be preferentially amplified.

### 44 Sebastes sequences amplified - 36 unique species
```{r}
sebastes.taxid = ecofind(taxo, '^Sebastes$')
is_a_rockfish=is.subcladeof(taxo,fish$taxid,sebastes.taxid)
table(is_a_rockfish)

setwd("//Users/Markus/Documents/UCLA/Barber Lab/Honor's Thesis/ecoPCR_results")
primer37 = read.ecopcr.result("primer37_10.31.18.ecoPCR")
sebastes.pcr.37 = subset(primer37,genus == 34820)
sebastes.species.37=unique(sebastes.pcr.37$species_name)
ss37df = as.data.frame(sebastes.species.37)
length(ss37df$sebastes.species.37)
```

### Resolution for Sebastes
```{r}
only.sebastes=fish[is_a_rockfish,]
res = resolution(taxo,only.sebastes)
resolution = with(only.sebastes,
                  unique(data.frame(species_name,taxid,rank=res)))
t(t(sort(table(resolution$rank)/length(resolution$rank),decreasing = TRUE)))

```

### Resolution for Fish outside of Sebastes
```{r}
no.sebastes=fish[is_a_fish & !is_a_rockfish,]
res = resolution(taxo,no.sebastes)
resolution = with(no.sebastes,
                  unique(data.frame(species_name,taxid,rank=res))
)
t(t(sort(table(resolution$rank)/length(resolution$rank),decreasing = TRUE)))
```

### Mismatch Plot
##### Degree of mismatches influence taxonomical selection - far more mismatches for fish outside of sebastes
```{r}
group=rep('vertebrate',length(is_a_fish))
group[is_a_fish]='fish'
group[is_a_rockfish]='Sebastes'
group=as.factor(group)
table(group)
par(mfcol=c(1,1))
mismatchplot(fish,group = group)
```

The x-axis is the number of mismatches on the forward primer and the y-axis is the number of mismatches on the reverse primer. This demonstrates the same relationship as the previous plots which showed the conservation of the primers, where the primers are far more specific for Sebastes than they are for fish outside of Sebastes.